### PR TITLE
chore: updated dependencies & configuration management

### PR DESCRIPTION
**Pull request by bloodf**
## Description
- created a dynamic `nuxt.config.js` configuration file based on you configuration, to enable `cloudnary` or `recaptcha`

`eslint`
- updated `eslint-plugin-unicorn` to `40.1.0`

`eslint-jest`
- updated `eslint-plugin-jest` to `26.0.0`

`eslint-typescript`
- `@typescript-eslint/eslint-plugin` to `5.10.2`
- `@typescript-eslint/parser` to `5.10.2`

`eslint-vue`
- updated `eslint-plugin-vue` to `8.4.0`

`api-client`
- updated `@apollo/client` to `3.5.8`
- updated `graphql` to `16.3.0`
- updated `cross-fetch` to `3.1.5`
- updated `typescript` to `4.5.5`
- updated `dotenv` to `15.0.0`
- updated `msw` to `0.36.8`

`theme`
- updated `isomorphic-dompurify` to `0.18.0`
- updated `@babel/core` to `7.16.12`
- updated `cypress` to `9.3.1`
- updated `dotenv` to `14.3.2`
- updated  `lint-staged` to `12.3.3`
- updated `mocha` to `9.2.0`
- updated `npm-check-updates` to `12.2.1`
- updated `typescript` to `4.5.5`

`repository`
- updated `@babel/core` to `7.16.12`
- updated `@babel/plugin-proposal-private-methods` to `7.16.11`
- updated `@commitlint/cli` to `16.1.0`
- updated `@graphql-codegen/cli` to `2.4.0`
- updated `@types/node` to `17.0.13`
- updated `@typescript-eslint/parser` to `5.10.1`
- updated `eslint` to `8.8.0`
- updated `npm-check-updates` to `12.2.1`
- updated `rollup` to `2.66.1`
- updated `vue-eslint-parser` to `8.2.0`
- updated  `@graphql-codegen/typescript-operations` to `2.2.3`
- updated  `@types/node` to `17.0.14`
- updated  `@typescript-eslint/parser` to `5.10.2`
- updated  `lint-staged` to `12.3.3`

## Motivation and Context
Update the packages and make it easier to enable dynamic content based on configuration

## How Has This Been Tested?
locally tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/deploy-vue-storefront-cloud.yml [hunk 1]
```diff
@@ -41,6 +41,7 @@ jobs:
           REDIS__HOST: ${{ secrets.REDIS__HOST }}
           REDIS__PORT: ${{ secrets.REDIS__PORT }}
           REDIS__KEY_PREFIX: ${{ secrets.REDIS__KEY_PREFIX }}
+          REDIS__PASSWORD: ${{ secrets.REDIS__PASSWORD }}
```
Comment (Frodigo) [inline .github/workflows/deploy-vue-storefront-cloud.yml:44]: REDIS_PASSWORD is not necessary for our cloud. Redis is enabled only in the scope of the docker container as DevOps said.

File: .vuestorefrontcloud/docker/Dockerfile [hunk 2]
```diff
@@ -15,6 +15,7 @@ ARG IMAGE_PROVIDER
 ARG IMAGE_PROVIDER_BASE_URL
 ARG REDIS__HOST
 ARG REDIS__PORT
+ARG REDIS__PASSWORD
```
Comment (Frodigo) [inline .vuestorefrontcloud/docker/Dockerfile:18]: As I mentioned above, please remove this arg also.

File: .vuestorefrontcloud/docker/Dockerfile [hunk 3]
```diff
@@ -42,6 +43,7 @@ ENV IMAGE_PROVIDER=${IMAGE_PROVIDER}
 ENV IMAGE_PROVIDER_BASE_URL=${IMAGE_PROVIDER_BASE_URL}
 ENV REDIS__HOST=${REDIS__HOST}
 ENV REDIS__PORT=${REDIS__PORT}
+ENV REDIS__PASSWORD=${REDIS__PASSWORD}
```
Comment (Frodigo) [inline .vuestorefrontcloud/docker/Dockerfile:46]: Please remove

File: .vuestorefrontcloud/docker/Dockerfile [hunk 4]
```diff
@@ -69,9 +71,7 @@ RUN mv /var/www/packages/theme/nuxt.config.js /var/www/packages/theme/base.nuxt.
 
 RUN yarn install
 
-RUN npx yarn@1.19.0 workspace @vue-storefront/magento-theme add @vsf-enterprise/redis-cache
-
-RUN npx yarn@1.19.0 workspace @vue-storefront/magento-theme add @nuxtjs/recaptcha
+RUN npx yarn@1.19.0 workspace @vue-storefront/magento-theme add @sentry/tracing @nuxtjs/sentry @vsf-enterprise/redis-cache
```
Comment (Frodigo) [inline .vuestorefrontcloud/docker/Dockerfile:74]: good solution, thanks!

File: packages/theme/.env.example [hunk 5]
```diff
@@ -9,9 +9,6 @@ MAGENTO_EXTERNAL_CHECKOUT_SYNC_PATH=/vue/cart/sync
 MAGENTO_BASE_URL={YOUR_SITE_FRONT_URL}
 IMAGE_PROVIDER={YOUR_IMAGE_PROVIDER}
 IMAGE_PROVIDER_BASE_URL={YOUR_IMAGE_PROVIDER_BASE_URL}
-MAGENTO_BASE_URL={YOUR_SITE_FRONT_URL}
```
Comment (Frodigo) [inline packages/theme/.env.example:12]: Can explain why you remove it?

**Review by bartoszherba — APPROVED** [review]

**Review by Frodigo — CHANGES_REQUESTED** [review]

File: packages/theme/.env.example (hunk 5 above)
Comment (bloodf) [reply packages/theme/.env.example:12]: duplicate :)

File: .vuestorefrontcloud/docker/Dockerfile (hunk 4 above)
Comment (bloodf) [reply .vuestorefrontcloud/docker/Dockerfile:74]: :)

File: .github/workflows/deploy-vue-storefront-cloud.yml (hunk 1 above)
Comment (bloodf) [reply .github/workflows/deploy-vue-storefront-cloud.yml]: done ✅

**Review by Frodigo — APPROVED** [review]

**Pull request merged**